pkg/controller/acmechallenges/sync_test.go

```diff
@@ -473,6 +473,9 @@ func TestSyncHappyPath(t *testing.T) {
 func runTest(t *testing.T, test testT) {
 	test.builder.T = t
 	test.builder.Init()
+	// RestConfig should be nil so to not attempt to call a real server over
+	// network.
+	test.builder.RESTConfig = nil
```

Comment:
Thanks @wallrj, indeed that is what is going on here.

I'm going to go with the `builder.InitWithRESTConfig()` approach since that seems the cleanest to me.